numpyro/distributions/directional.py

```diff
@@ -0,0 +1,120 @@
+import jax.numpy as np
+from jax import lax, random
+
+from numpyro.distributions import constraints
+from numpyro.distributions.distribution import Distribution
+from numpyro.distributions.util import validate_sample
+from numpyro.util import copy_docs_from
+
+
+@copy_docs_from(Distribution)
+class VonMises(Distribution):
+    arg_constraints = {'location': constraints.real, 'concentration': constraints.positive}
+    support = constraints.real
+    s_cutoff_map = {np.dtype(np.float16): 1.8e-1,
+                    np.dtype(np.float32): 2e-2,
+                    np.dtype(np.float64): 1.2e-4}
```

Comment:
Could you move this to `_sample_centered` implementation? Those `np.` ops will trigger GPU backend when import numpyro.

Reply:
I'll move it in and add a reference. The numbers are from the TensorFlow implementation, which may be the cause of the difference.